src/scanRegistration.cpp

```diff
@@ -384,25 +492,33 @@ void laserCloudHandler(const sensor_msgs::PointCloud2ConstPtr& laserCloudIn2)
                 + laserCloud->points[i + 3].z + laserCloud->points[i + 4].z
                 + laserCloud->points[i + 5].z;
     
+    // calculate smoothness
     laserCloud->points[i].s = diffX * diffX + diffY * diffY + diffZ * diffZ;
   }
   
+  float m = 0;
   for (int i = 5; i < cloudSize - 6; i++) {
+    // calculate difference with the neighbour
     float diffX = laserCloud->points[i + 1].x - laserCloud->points[i].x;
     float diffY = laserCloud->points[i + 1].y - laserCloud->points[i].y;
     float diffZ = laserCloud->points[i + 1].z - laserCloud->points[i].z;
     float diff = diffX * diffX + diffY * diffY + diffZ * diffZ;
 
+    // incremental average calculation of the diff
+    m = m + ((diff-m)/(cloudSize-12));
```

Comment:
This was my addition to calculate incremental average in order to see what is a "normal" value for the diff. Need to clean it up once I'm done with the IMU behaviour (a simple rotation was not enough for the IMU, still debugging).